imagetest/test_suites/packagevalidation/package_test.go

```diff
@@ -38,41 +61,74 @@ func TestStandardPrograms(t *testing.T) {
 func TestGuestPackages(t *testing.T) {
 	utils.LinuxOnly(t)
 	image, err := utils.GetMetadata(utils.Context(t), "instance", "image")
+
 	if err != nil {
 		t.Fatalf("couldn't determine image from metadata")
 	}
-	cmdPrefix := []string{"rpm", "-q", "--queryformat", "'%{NAME}\n'"}
+
+	// What command to list all packages
+	listPackagesCmd := []string{"rpm", "-qa", "--queryformat", "%{NAME}\n"}
 	if strings.Contains(image, "debian") || strings.Contains(image, "ubuntu") {
-		cmdPrefix = []string{"dpkg-query", "-W", "--showformat", "'${Package}\n'"}
+		listPackagesCmd = []string{"dpkg-query", "-W", "--showformat", "${Package}\n"}
+	}
+
+	// What packages to insure are not installed
+	excludePkgs := []string{}
+
+	// What packages to check for and excude for diffrent images
+	requiredPkgs := []string{"google-guest-agent", "google-osconfig-agent"}
+	if strings.Contains(image, "rhel") {
+		requiredPkgs = append(requiredPkgs, "google-compute-engine", "gce-disk-expand", "google-cloud-sdk")
 	}
-	packages := []string{"google-guest-agent", "google-osconfig-agent"}
-	if strings.Contains(image, "sles") {
-		packages = append(packages, "google-guest-configs") // SLES name for 'google-compute-engine' package.
-		packages = append(packages, "google-guest-oslogin")
-	} else {
-		packages = append(packages, "google-compute-engine")
-		packages = append(packages, "google-compute-engine-oslogin")
+	if strings.Contains(image, "sles") || strings.Contains(image, "suse") {
+		requiredPkgs = append(requiredPkgs, "google-guest-configs") // SLES name for 'google-compute-engine' package.
+		requiredPkgs = append(requiredPkgs, "google-guest-oslogin")
 	}
 	if strings.Contains(image, "centos-7") {
-		packages = append(packages, "epel-release")
-	}
-
-	for _, pkg := range packages {
-		args := append(cmdPrefix[1:], pkg)
-		cmd := exec.Command(cmdPrefix[0], args...)
-		stderr := &bytes.Buffer{}
-		cmd.Stderr = stderr
-		stdout := &bytes.Buffer{}
-		cmd.Stdout = stdout
-		if err := cmd.Start(); err != nil {
-			t.Errorf("error starting check command: %v", err)
-			continue
+		requiredPkgs = append(requiredPkgs, "epel-release")
+	}
+	if strings.Contains(image, "debian") {
+		requiredPkgs = append(requiredPkgs, "haveged", "net-tools")
+		requiredPkgs = append(requiredPkgs, "google-cloud-packages-archive-keyring", "isc-dhcp-client")
+		requiredPkgs = append(requiredPkgs, "google-compute-engine", "gce-disk-expand", "python3", "google-cloud-cli")
```

Comment:
I think checking for `python3` is a mistake, we don't really care about python we care that the gcloud cli is installed and working which is covered by everything else in this suite. In some hypothetical `python4` future, we don't really care which version gcloud uses as long as it works.

Reply:
true we explicitly install it, but if the gcloud cli is working its fine, no need to duplicate